File: VB/Q234787/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace Q234787
    Partial Public Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub dateEdit1_Properties_CustomWeekNumber(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Repository.CustomWeekNumberEventArgs) Handles dateEdit1.Properties.CustomWeekNumber
            Dim [date] As Date = e.Date
            Dim baseDate As Date = If([date].Month < 10, New Date([date].Year - 1, 10, 1), New Date([date].Year, 10, 1))
            e.WeekNumber = CInt((Math.Floor(CDbl((New Date([date].Ticks - baseDate.Ticks)).DayOfYear \ 7))))
        End Sub
    End Class
End Namespace

File: CS/Q234787/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace Q234787 {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
        }  

        private void dateEdit1_Properties_CustomWeekNumber(object sender, DevExpress.XtraEditors.Repository.CustomWeekNumberEventArgs e) {
            DateTime date = e.Date;
            DateTime baseDate = date.Month < 10 ? new DateTime(date.Year - 1, 10, 1) : new DateTime(date.Year, 10, 1);
            e.WeekNumber = (int)Math.Floor((double)(new DateTime(date.Ticks - baseDate.Ticks).DayOfYear / 7));
        }
    }
}